coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -656,6 +696,34 @@ func (server *Server) CompleteJob(ctx context.Context, completed *proto.Complete
 			return nil, xerrors.Errorf("complete job: %w", err)
 		}
 
+		// audit the outcome of the workspace build
+		if getWorkspaceError == nil {
+			auditor := server.Auditor.Load()
+			auditAction := determineAuditAction(workspaceBuild.Transition)
+
+			// We pass the workspace name to the Auditor so that it
+			// can form a friendly string for the user.
+			workspaceResourceInfo := map[string]string{
+				"workspaceName": workspace.Name,
+			}
+
+			wriBytes, err := json.Marshal(workspaceResourceInfo)
+			if err != nil {
+				server.Logger.Error(ctx, "could not marshal workspace name", slog.Error(err))
+			}
```

Comment:
I would have an else here to set `wriBytes` to `[]byte("{}")`, otherwise auditing it could fail.

Reply:
@coadler does [this check](https://github.com/coder/coder/pull/5143/files#diff-cc84da01e587fa9b8971d66c92f68321bad61d76a161a36926ccfe6d57277048R191) in the audit function itself not cover that scenario?

Reply:
Yeah it does! Didn't catch that